Functions/Administration/xASL_adm_DefineASLSequence.m

```diff
@@ -27,6 +27,21 @@
     warning('x.Q.readoutDim parameter missing');
 end
 
+% Check for illegal sequence definitions
+if isfield(x.Q,'Sequence')
+    if isempty(regexpi(x.Q.Sequence, '(2d_epi|3d_grase|3d_spiral)'))
+        warning('Unknown ASL sequence');
+
+        if strcmpi(x.Q.readoutDim,'2D')
```

Comment:
You can improve this. If the readout is 3D then Philips and Siemens have 3D GRASE and GE will have 3D spiral ;)
But please also build a  isfield check for vendor.

Reply:
That is already there below, where we indeed also check for the vendor.
Here, we already assume 2D EPI even if there is no vendor specified.

But you're right that we should still do this check even if an unknown ASL sequence was detected.

Reply:
Good point to delete the unknown sequence. And you are right that we assign the new sequence below. But actually, now we don't need to assign 2D_EPI here as this is done below. So we should remove 2D_EPI from here and only keep it below.

Reply:
Not true, because we run the code below only if we know the vendor. Currenty, even if we don't know the vendor, we can assign 2D EPI. I will now delete this below so there is no confusion.

Reply:
Yes - I know, it needed an extra edit ;)